### PR TITLE
Update data API ref for v0.3.0 release

### PR DESCRIPTION
**Pull request by joecummings**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
* Update data API ref for v0.3.0 release

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.)

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [x] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Example of docstring: https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285
Example in our docs: https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models

- [ ] I did not change any public API;
- [ ] I have added an example to docs or docstrings;

### CODE REVIEW

File: torchtune/data/_collate.py [hunk 1]
```diff
@@ -227,16 +227,21 @@ def padded_collate_tiled_images_and_mask(
     """Pad a batch of text sequences, tiled image tensors, aspect ratios,
     and cross attention masks. This can be used for both training and inference.
 
-    ``batch`` is expected to be a list of sample dicts containing the following:
+    ``batch`` is expected to be a list of sample dicts containing the following::
         - "tokens": List[int] of length text_seq_len, varies across samples
         - "labels": List[int] of length text_seq_len, varies across samples
         - "encoder_input": Dict[str, List[torch.Tensor]]
             - "images": List[torch.Tensor], each with shape (n_tiles, c, h, w)
             - "aspect_ratio": List[torch.Tensor], each with shape (2, ) to indicate h_ratio, w_ratio
```
Comment (ebsmothers) [inline torchtune/data/_collate.py:235]: ``Needs`` ``more`` ``backticks``

**Review by ebsmothers — APPROVED** [review]
